examples/asr_training/rnnt_lightning.py

```diff
@@ -0,0 +1,535 @@
+                ),
+                CustomDataset(
+                    torchaudio.datasets.LIBRISPEECH(
+                        self.librispeech_path, url="train-other-500"
+                    ),
+                    1000,
+                ),
+            ]
+        )
+        dataloader = torch.utils.data.DataLoader(
+            dataset,
+            batch_size=None,
+            collate_fn=self._train_collate_fn,
+            num_workers=10,
+            shuffle=True,
+        )
+        return dataloader
+
+    def val_dataloader(self):
+        dataset = torch.utils.data.ConcatDataset(
+            [
+                CustomDataset(
+                    torchaudio.datasets.LIBRISPEECH(
+                        self.librispeech_path, url="dev-clean"
+                    ),
+                    1000,
+                ),
+                CustomDataset(
+                    torchaudio.datasets.LIBRISPEECH(
+                        self.librispeech_path, url="dev-other"
+                    ),
+                    1000,
+                ),
+            ]
+        )
+        dataloader = torch.utils.data.DataLoader(
+            dataset, batch_size=None, collate_fn=self._valid_collate_fn, num_workers=10,
+        )
+        return dataloader
+
+    def test_dataloader(self):
+        dataset = torchaudio.datasets.LIBRISPEECH(
+            self.librispeech_path, url="test-clean"
+        )
+        dataloader = torch.utils.data.DataLoader(
+            dataset, batch_size=1, collate_fn=self._test_collate_fn
+        )
+        return dataloader
+
+
+def compute_word_level_distance(seq1, seq2):
+    return torchaudio.functional.edit_distance(
+        seq1.lower().split(), seq2.lower().split()
+    )
+
+
+def post_process_hypos(
+    hypos: List[Hypothesis], sp_model: spm.SentencePieceProcessor, tgt_dict: Dictionary
+) -> List[Tuple[str, float, List[int], List[int]]]:
+    post_process_remove_list = [
+        sp_model.unk_id(),
+        sp_model.eos_id(),
+        sp_model.pad_id(),
+    ]
+    hypos_str = [
+        tgt_dict.string(
+            [
+                token_index
+                for token_index in h.tokens[1:]
+                if token_index not in post_process_remove_list
+            ]
+        )
+        for h in hypos
+    ]
+    hypos_str = [sp_model.DecodePieces(s.split()) for s in hypos_str]
+    hypos_ali = [h.alignment[1:] for h in hypos]
+    hypos_ids = [h.tokens[1:] for h in hypos]
+    hypos_score = [[math.exp(h.score)] for h in hypos]
+
+    nbest_batch = list(zip(hypos_str, hypos_score, hypos_ali, hypos_ids))
+
+    return nbest_batch
+
+
+def run_eval(args):
+    model = (
+        RNNTModule.load_from_checkpoint(
+            args.checkpoint_path,
+            librispeech_path=args.librispeech_path,
+            sp_model_path=args.sp_model_path,
+            tgt_dict_path=args.tgt_dict_path,
+            global_stats_path=args.global_stats_path,
+        )
+        .eval()
+        .to(device="cuda")
```

Comment:
This could cause error if user doesn't have cuda device. Add `num_gpu` or `use_cuda` argument in the parser can help it.